src/v/ssx/tests/async_algorithm_test.cc

```diff
@@ -98,75 +103,110 @@ struct task_counter {
 
 } // namespace
 
-TEST(AsyncAlgo, async_for_each_same_result) {
+std::vector<int> make_container(size_t elems, std::vector<int>) {
+    std::vector<int> ret;
+    for (int i = 0; i < elems; i++) {
+        ret.push_back(i);
+    }
+    return ret;
+}
+
+std::unordered_map<int, int>
+make_container(size_t elems, std::unordered_map<int, int>) {
+    std::unordered_map<int, int> ret;
+    for (int i = 0; i < elems; i++) {
+        ret[i] = i;
+    }
+    return ret;
+}
+
+template<typename T>
+struct AsyncAlgo : public testing::Test {
+    using container = T;
+
+    static T make(size_t size) { return make_container(size, T{}); }
+};
+
+// using container_types = ::testing::Types<std::vector<int>>;
+using container_types
+  = ::testing::Types<std::vector<int>, std::unordered_map<int, int>>;
```

Comment:
Make all tests except one execute for both vector and unordered_map (latter of which has only a forward iterator).